src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,73 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { FilteringType } from '../../types/FilteringType';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  input: string
+  setInput: React.Dispatch<React.SetStateAction<string>>
+  selectedButton: string
+  setSelectedButton: React.Dispatch<React.SetStateAction<FilteringType>>
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  input,
+  setInput,
+  selectedButton,
+  setSelectedButton,
+}) => {
+  function handleInput(e: React.BaseSyntheticEvent) {
+    setInput(e.target.value);
+  }
+
+  const handleSelect = (value: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedButton(value.target.value as FilteringType);
+  };
+
+  const handleDeleteButton = () => {
+    setInput('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={selectedButton}
+            onChange={(value) => handleSelect(value)}
```

Comment:
```suggestion
            onChange={handleSelect}
```
If a mentor points out a mistake, it means that you need to fix ALL occurrences of the